Mage/src/main/java/mage/players/PlayerImpl.java

```diff
@@ -2881,6 +2881,22 @@ public boolean seekCard(FilterCard filter, Ability source, Game game) {
         return true;
     }
 
+    @Override
+    public boolean putRandomCardInGraveyard(FilterCard filter, Ability source, Game game) {
```

Comment:
I don't agree with putting this as inner method in `Player` classes. It's only used in two cards as far as I can tell? What's wrong with putting the code only in the custom card class where it is used?

Reply:
Agreeing with the sentiment.
Either in the custom Effect, or in a shared Effect that could handle all those: https://scryfall.com/search?q=o%3A%22put+a+random%22&unique=cards&as=grid&order=name